awscli/customizations/s3/filegenerator.py

```diff
@@ -304,10 +302,8 @@ def _list_single_object(self, s3_path):
         # IAM policies with the smallest set of permissions needed) and
         # instead use a HeadObject request.
         bucket, key = find_bucket_key(s3_path)
-        operation = self._service.get_operation('HeadObject')
         try:
-            response = operation.call(
-                self._endpoint, bucket=bucket, key=key)[1]
+            response = self._client.head_object(Bucket=bucket, Key=key)
         except ClientError as e:
             # We want to try to give a more helpful error message.
             # This is what the customer is going to see so we want to
```

Comment:
Please double check if we need all this still.  The client objects should handle these scenarios for us, so we shouldn't need to fix up error messages here.